boot/zephyr/main.c

```diff
@@ -371,30 +371,70 @@ void zephyr_boot_log_stop(void)
         */
 
 #if defined(CONFIG_MCUBOOT_SERIAL) || defined(CONFIG_BOOT_USB_DFU_GPIO)
-static bool detect_pin(const char* port, int pin, uint32_t expected, int delay)
+
+#ifdef CONFIG_MCUBOOT_SERIAL
+#define BUTTON_0_DETECT_DELAY CONFIG_BOOT_SERIAL_DETECT_DELAY
+#else
+#define BUTTON_0_DETECT_DELAY CONFIG_BOOT_USB_DFU_DETECT_DELAY
+#endif
+
+
+#define BUTTON_0_NODE DT_ALIAS(mcuboot_button0)
+
+#if DT_NODE_HAS_STATUS(BUTTON_0_NODE, okay) && DT_NODE_HAS_PROP(BUTTON_0_NODE, gpios)
```

Comment:
Why not just use `DT_NODE_EXISTS`? Status is not usually checked (nor set) for button and LED nodes. Sure, the above will work fine, but it looks inconsistent with other pieces of code that use a button.

```suggestion
#if DT_NODE_EXISTS(BUTTON_0_NODE) && DT_NODE_HAS_PROP(BUTTON_0_NODE, gpios)
```

Reply:
fixed